### PR TITLE
Validate compose and setupwizard schemas

### PR DESCRIPTION
**Pull request by pablomendezroyo**
In order to have a more resilient and consistent dappnodeSDK this PR performs the schema validation for compose and setupwizard

Changes:
- [x] Add and relocate compose, setupwizard and manifest types
- [x] Add compose schema
- [x] Validate compose and setupwizard schemas
- [x] Add validation schemas tests for setupwizard and compose

### CODE REVIEW

File: src/schemaValidation/validateSchema.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+const ajv = new Ajv({
+  allErrors: true,
+  coerceTypes: true,
+  strictSchema: false
+});
+
+ajvErrors(ajv);
+
+console.log("Validating manifest schema");
```
Comment (dapplion) [inline src/schemaValidation/validateSchema.ts:21]: Why are this logs here? Doesn't this log always in response to any command?

File: src/schemaValidation/validateSchema.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+const manifestValidate = ajv.compile(manifestSchema);
```
Comment (dapplion) [inline src/schemaValidation/validateSchema.ts:22]: I think ajv.complie caches the result. Doing it here will require compiling schemas if you just want to --help. I think it's okay to do it on demand

File: src/schemaValidation/validateSchema.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+/**
+ * Validates a manifest syncronously. Does NOT throw.
+ * @param manifest
+ * @returns = {
+ *   valid: false|true
+ *   errors: [
+ *     "manifest should have required property 'description'",
+ *     "manifest.image.path should be an non-empty string",
+ *   ]
+ * }
```
Comment (dapplion) [inline src/schemaValidation/validateSchema.ts:37]: It returns void or an object?

File: src/schemaValidation/validateSchema.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+ */
+export function validateReleaseFile(
```
Comment (dapplion) [inline src/schemaValidation/validateSchema.ts:62]: What's the point of multplexing arguments here with `inferReleaseFileType`? Why not just do:

```ts
function validateManifestSchema(manifest: Manifest)
function validateComposeSchema(compose: Compose)
function validateSetupWizardSchema(setupWizard: SetupWizard)

File: src/schemaValidation/validateSchema.ts [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+function inferReleaseFileType(
+  releaseFile: Manifest | Compose | SetupWizard
+): ReleaseFileType {
+  if ("name" in releaseFile) {
+    return "manifest";
+  } else if ("services" in releaseFile) {
+    return "compose";
+  } else if ("fields" in releaseFile) {
+    return "setupWizard";
+  } else throw Error("Unknown release file type");
```
Comment (dapplion) [inline src/schemaValidation/validateSchema.ts:104]: This function is super dangerous, and not worth the trouble. There may be key collisions in the future

File: src/tasks/buildAndUpload.ts [hunk 6]
```diff
@@ -168,25 +175,47 @@ as ${releaseFilesDefaultNames.avatar} and then remove the 'manifest.avatar' prop
...
     },
 
+    {
+      title: "Validate files",
+      task: () => {
+        for (const [fileId] of Object.entries(releaseFiles)) {
+          switch (fileId as keyof typeof releaseFiles) {
+            case "setupWizard":
+              if (!setupWizardString) continue;
+              validateSchema(yaml.load(setupWizardString));
```
Comment (dapplion) [inline src/tasks/buildAndUpload.ts:185]: What the reason for this? I don't follow

File: src/schemaValidation/validateSchema.ts (hunk 1 above)
Comment (pablomendezroyo) [reply src/schemaValidation/validateSchema.ts:21]: ups, I was debugging, forgot to delete it

File: src/tasks/buildAndUpload.ts (hunk 6 above)
Comment (pablomendezroyo) [reply src/tasks/buildAndUpload.ts]: fixed. Read in the block scope and then dropped

File: src/schemaValidation/validateSchema.ts (hunk 5 above)
Comment (pablomendezroyo) [reply src/schemaValidation/validateSchema.ts]: removed

File: src/schemaValidation/validateSchema.ts (hunk 2 above)
Comment (pablomendezroyo) [reply src/schemaValidation/validateSchema.ts]: makes sense, now compiling on demmand

File: test/utils/versions/increaseFromLocalVersion.test.ts [hunk 7]
```diff
@@ -20,7 +20,8 @@ describe("increaseFromLocalVersion", function () {
   this.timeout(60 * 1000);
 
   const dnpName = "admin.dnp.dappnode.eth";
-  const manifest = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const manifest: any = {
```
Comment (dapplion) [inline test/utils/versions/increaseFromLocalVersion.test.ts:24]: Why type any?

File: test/validate.test.ts [hunk 8]
```diff
@@ -29,13 +28,12 @@ describe("utils / format", () => {
...
-    const validManifest = validateManifestSchema(manifest);
-    expect(validManifest.valid).to.be.true;
-    expect(validManifest.errors).to.be.empty;
+    expect(() => validateManifestSchema(manifest)).to.not.throw();
   });
 
   it("throw error validating", () => {
-    const manifest = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const manifest: any = {
```
Comment (dapplion) [inline test/validate.test.ts:36]: Why type any?

File: test/utils/versions/increaseFromLocalVersion.test.ts (hunk 7 above)
Comment (pablomendezroyo) [reply test/utils/versions/increaseFromLocalVersion.test.ts:24]: omit invalid value to cast the type to a valid `Manifest`

**Review by dapplion — APPROVED** [review]
LGTM!

**Pull request merged**